cis194/week6/tien/src/Fibonacci.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE FlexibleInstances #-}
+module Fibonacci where
+
+fib :: Integer -> Integer
+fib 0 = 0
+fib 1 = 1
+fib n = fib (n-1) + fib (n-2)
+
+-- exercise 1 --
+fibs1::[Integer]
+fibs1 = map fib [0..]
```

Comment:
The use of `fmap` or the infix version `<$>` is preferred as we do not need to change the implementation if we change `[Integer]` to any `Functor f => f Integer`

```suggestion
fibs1 = fmap fib [0..]
```

or

```suggestion
fibs1 = fib <$> [0..]
```